applications/tests/test_applications_new_schema_mutations.py

```diff
@@ -61,39 +55,37 @@ def test_update_berth_application(superuser, berth_application, customer_profile
     }
 
 
-def test_update_berth_application_no_application_id(superuser, customer_profile):
+def test_update_berth_application_no_application_id(
+    superuser_api_client, customer_profile
+):
     variables = {
         "customerId": to_global_id("BerthProfileNode", str(customer_profile.id)),
     }
 
-    executed = client.execute(
-        query=UPDATE_BERTH_APPLICATION_MUTATION,
-        variables=variables,
-        graphql_url=GRAPHQL_URL,
-        user=superuser,
+    executed = superuser_api_client.execute(
+        UPDATE_BERTH_APPLICATION_MUTATION, input_variables=variables,
     )
 
     assert_field_missing("id", executed)
 
 
-def test_update_berth_application_no_customer_id(superuser, berth_application):
+def test_update_berth_application_no_customer_id(
+    superuser_api_client, berth_application
+):
     variables = {
         "id": to_global_id("BerthApplicationNode", str(berth_application.id)),
     }
 
-    executed = client.execute(
-        query=UPDATE_BERTH_APPLICATION_MUTATION,
-        variables=variables,
-        graphql_url=GRAPHQL_URL,
-        user=superuser,
+    executed = superuser_api_client.execute(
+        UPDATE_BERTH_APPLICATION_MUTATION, input_variables=variables,
     )
 
     assert_field_missing("customerId", executed)
 
 
-@pytest.mark.parametrize("user", ["none", "base", "staff"], indirect=True)
+@pytest.mark.parametrize("user", [None, "user", "staff"])
```

Comment:
By replacing the parametrising with this:
```suggestion
@pytest.mark.parametrize(
    "api_client", ["api_client", "user_api_client", "staff_api_client"], indirect=True
)
```

we could avoid having to manually ask for the fixture. I feel like it's an unnecessary step considering that there's already a way to do it and less hacky.

```python
    if user:
        api_client = request.getfixturevalue("{}_api_client".format(user))
```

I agree that parametrising specifically user might make it a bit clearer, but I'm not 100% convinced on manually getting the fixtures.

Reply:
yeah, I spent some time debating in my head whether it's a good idea or not to do `getfixturevalue` 😅 I'll implement it the way you suggest for a couple of tests to see how it behaves and looks

Reply:
ok, I'll switch to that 👍